src/synapse/api.py

```diff
@@ -461,3 +467,25 @@ def _do_request(
         if "User ID already taken" in exc.response.text:
             raise UserExistsError("User exists") from exc
         raise NetworkError(f"HTTP error from {url}.") from exc
+
+
+def change_user_admin(
+    user: User,
+    server: typing.Optional[str],
+    admin_access_token: typing.Optional[str],
+) -> None:
+    """Change user to admin.
+
+    Args:
+        user: user to be changed to admin.
+        server: to be used to create the user id.
+        admin_access_token: server admin access token to be used.
+    """
+    authorization_token = f"Bearer {admin_access_token}"
+    headers = {"Authorization": authorization_token}
```

Comment:
This should be in a private method of the module, this is used multiple times (and will probably be used if the module needs more api calls)
This could also be added to `_do_requests` by adding a `requires_authentication` boolean argument

Reply:
Nice suggestion, changed.